Community-Supported/flights-data-incremental-refresh/flights-data-incremental-refresh.py

```diff
@@ -0,0 +1,118 @@
+                ])
+            # Create the flights table.
+            connection.catalog.create_table(table_definition)
+
+            # Insert each of the states into the table.
+            with Inserter(connection, table_definition) as inserter:
+                for s in states.states:
+                    inserter.add_row([s.baro_altitude, s.callsign, s.latitude, s.longitude, s.on_ground, s.origin_country, s.time_position, s.velocity])
+                inserter.execute()
+
+            num_flights = connection.execute_scalar_query(query=f"SELECT COUNT(*) from {table_definition.table_name}")
+            print(f"Inserted {num_flights} flights into {database_path}.")
+
+def publish_to_server(server_url, tableau_auth, project_name, database_path, datasource_name_on_server):
+    """
+    Creates the datasource on Tableau Server if it has not yet been created. Otherwise, uses the 
+    Hyper Update REST API (https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_how_to_update_data_to_hyper.htm) to append the data to the datasource.
+    """
+    # Create a tableuserverclient object to interact with Tableau Server.
+    server = TSC.Server(server_url, use_server_version=True)
+    # Sign into Tableau Server with the above authentication information.
+    with server.auth.sign_in(tableau_auth):
+        # Get project_id from project_name.
+        matching_projects = server.projects.filter(name=project_name)
+        project_id = next((project.id for project in matching_projects if project.name == project_name), None)
+        if project_id is None:
+            print(f"Publish failed. The specified project '{project_name}' does not exist.")
+            exit()
+
+        # Get the datasource from Server (if it exists).
+        matching_datasources = server.datasources.filter(name=datasource_name_on_server)
+        datasource = next((ds for ds in matching_datasources), None)
+
+        if datasource is None:
+            # If the datasource does not exist on server, publish the datasource. 
+            publish_mode = TSC.Server.PublishMode.CreateNew
+            datasource = TSC.DatasourceItem(project_id)
+            # Set the name of the datasource such that it can be easily identified.
+            datasource.name = datasource_name_on_server
+            datasource = server.datasources.publish(datasource, database_path, publish_mode)
+            print(f"New datasource published: (id : {datasource.id}, name: {datasource.name}).")
+        else:
+            # If the datasource already exists on Tableau Server, use the Hyper Update REST API 
+            # to send the delta to Tableau Server and insert the data into the respective table
+            # in the datasource. 
+
+            # Create a new random request id. 
+            request_id = str(uuid.uuid4())
+
+            # Create one action that inserts from the new table into the existing table.
```

Comment:
add link to https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_how_to_update_data_to_hyper.htm#action-batch-descriptions